starport/cmd/module-create.go

```diff
@@ -86,6 +90,35 @@ func createModuleHandler(cmd *cobra.Command, args []string) error {
 		options = append(options, scaffolder.WithIBCChannelOrdering(ibcOrdering), scaffolder.WithIBC())
 	}
 
+	// Get module dependencies
+	dependencies, err := cmd.Flags().GetStringSlice(flagDep)
+	if err != nil {
+		return err
+	}
+	if len(dependencies) > 0 {
+		var formattedDependencies []modulecreate.Dependency
+
+		// Parse the provided dependencies
+		for _, dependency := range dependencies {
+			splitted := strings.Split(dependency, ".")
```

Comment:
Can you please elaborate the usage in this: `starport module create bar --dep mybank.BankKeeper,account,foo`, for ex. is `mybank` package name for the module?

I don't see how we can both support the dotted dependencies pattern and not dotted ones together. The syntax should always be `mykeeper.MyKeeper` where `mykeeper` is the actual Go package name that `MyKeeper` struct is located at.

After receving this info from the `--dep` flag, we should analyse the App struct and find a field that has this  type. If we find multiple fields with the same type, we can return with an error. If we cannot find any it's better we stop the process and return with an error as well.

```
type App struct {
   myKeeperField mykeeper.MyKeeper
   ^
}
```

After, we need check app.go again to find out the variable name that an instance of App assigned to and if we find multiple, we need to return with an error.

```
var myapp = &App{}  
      ^
```

Then, we can use `myapp.myKeeperField`.

Note:

If we wish to support non dotted deps and provide a way to just pass the module name within `--dep`, we can do this for only sdk modules and popular modules that we want to support such as liquidity.

It should work in the same way with dotted version. But this time, Starport will internally know what is `mykeeper` and `MyKeeper` for a given module because we'll keep this as a list internally.

```
var  knownKeepers = map[string]string{
    "bank": "bankkeeper.Keeper",
}
```

Reply:
Ok, the dot is totally confusing as a separator (I didn't have the go syntax in mind).
This is here an optional separator that allows you to specify for your dependency, the name of the dependency keeper in app.go in case it's a custom one.
If you provide --dep foo.bar so the name of your dependency is foo. The foo instance in the module you generate will be newmodule.FooKeeper. bar is the name of the keeper instance in app.go, you will check for bar in app.go and provide app.bar to the new module constructor. If the keeper name is not provided we use the standardized one which is FooKeeper (<ModuleName>Keeper)
 
It's important to precise we don't do any type or module verification because there are none to do. We only check variable name because a module dependency is an empty interface, it can be anything.
 
For example, if you provide --dep bank, the bank keeper can be anything and can come from any module. You can potentially provide a foo.Keeper from github.com/foo/foo as the expected bank keeper, the way we get the correct instance name to provide to the module is by searching for an attribute in the app named BankKeeper, this is the attribute name that identifies it as the bank module.
 
The second optional value in the dot separator (this dot must be replaced) is only used so you explicitely tells what is the concrete keeper name you use in app.go. I'm also wondering if we shouldn't just remove this functionality because following the same reasoning from my comment above of the lazy check for App. It may be better to make the check very restrictive so we don't automatically register the module at the first sign the user changed app.go
 
It's definitely necessary to be able to provide the simple syntax --dep bank,account,foo (where foo is the name of a module previously scaffolded with Starport) because this is the benefit of this command, simplifying the usage of module dependency

Reply:
>After, we need to check app.go again to find out the variable name that an instance of App assigned to and if we find multiple, we need to return with an error.
var myapp = &App{}  
      ^

That can be an interesting check to perform before creating a module however it should definitely handled in another PR because it's an issue for module creation in a general point of view (currently the module creation would fails if you modify this instance name)

Reply:
I'm no sure if I get what you mean 100%, but I get the general sense. Please let me know if you need for me to elaborate suggestions from my previous comment. I believe we should require the dot syntax in general but make it optional for sdk and popular modules.

For the `var myapp = &App{}`, I prefer to avoid making big assumptions in the code base. I think we should include it to this PR in favor of not leaving technical depts, it is an easy implementation..

Reply:
> For the var myapp = &App{}, I prefer to avoid making big assumptions in the code base. I think we should include it to this PR in favor of not leaving technical depts, it is an easy implementation

This is not a question of technical debt. If `app` variable name is changed the module generation fails anyway with our current module scaffolding, managing `app` name changes is beyond the scope of the PR since it requires further modification not related to dependencies